File: evidence/generated-systems/golang/advertising-on-golang/advertising-on-golang/internal/model/DeviceCriterion.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// DeviceCriterion Declaration
//==============================================================
type DeviceCriterion struct {
    gorm.Model
     TargetingProfileId         *uint
    TargetingProfile           *TargetingProfile `gorm:"foreignKey:TargetingProfileId"`
    DeviceType                      DeviceType
    PlatformType                      PlatformType
    Operator                      TargetingOperator

// parent associations as their child

}

File: evidence/generated-systems/angular/advertising-on-angular/src/app/models/DeviceCriterion.js

// Define collection and schema for DeviceCriterion
export interface DeviceCriterion {
    TargetingProfile:
	type : Schema.Types.ObjectId
    DeviceType:
 	type : String
    PlatformType:
 	type : String
    Operator:
 	type : String
#
    collection: 'deviceCriterions'
}
